### PR TITLE
Add libsodium Public-key Signature Functionality

### PR DESCRIPTION
**Pull request by dorkrawk**
Added libsodium bindings for Public-key Signatures: https://download.libsodium.org/doc/public-key_cryptography/public-key_signatures.html

I needed this functionality for a project I'm working on and I thought it might be a nice expansion to this library.

Example usage:
```
message = "Hello World!"

signing_pair = Cox::SignKeyPair.new

# Sign the message
signature = Cox.sign_detached(message, signing_pair.secret)

# And verify
Cox.verify_detached(signature, message, signing_pair.public) # => true
```

### CODE REVIEW

**Comment by andrewhamon** [comment]
Hey, somehow missed this until now! Taking a look now.

File: src/cox/signature.cr [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require "./lib_sodium"
+
+module Cox
+  class Signature
```
Comment (andrewhamon) [inline src/cox/signature.cr]: I don't think this is being used at all is it? Can we delete it?

File: src/cox.cr [hunk 2]
```diff
@@ -22,6 +22,24 @@ module Cox
     LibSodium.crypto_box_open_easy(output_buffer.to_unsafe, data_buffer.to_unsafe, data_size, nonce.pointer, sender_public_key.pointer, recipient_secret_key.pointer)
     output_buffer
   end
+
+  def self.sign(message, secret_key : SignSecretKey)
```
Comment (andrewhamon) [inline src/cox.cr]: Could we rename these to `sign_detached` and `verify_detached`? That leaves room to non-disruptively add the combined version down the road :)

**Review by andrewhamon — CHANGES_REQUESTED** [review]
Thanks for the PR! Almost perfect just a few quibbles.

**Comment by dorkrawk** [comment]
Yep, I think those changes make total sense.  I've removed the `Signature` class and updated the method names.

**Review by andrewhamon — APPROVED** [review]
Thanks for the PR! Merging :D 

Sorry for the delay.

**Pull request merged**